File: android-native/app/src/main/java/com/jottery/android/model/ConflictInfo.kt
package com.jottery.android.model

/**
 * In-memory only — for conflict resolution UI.
 */
data class ConflictInfo(
    val id: String,
    val localContent: String,
    val localTags: List<String>,
    val localModifiedAt: String,
    val serverContent: String,
    val serverTags: List<String>,
    val serverModifiedAt: String,
    val serverEncryptedContent: String,
    val serverEncryptedTags: List<String>,
    val serverVersion: Int,
    val serverAttachments: List<AttachmentRef>,
    val serverPinned: Boolean,
    val serverSyntaxLanguage: String? = null,
    val serverWordWrap: Boolean? = null,
    val serverShowPreview: Boolean? = null,
    val serverContentHash: String? = null,
    val serverParentHash: String? = null,
    val serverHashChain: List<String>? = null,
)

enum class ConflictResolutionStrategy {
    KEEP_LOCAL,
    KEEP_SERVER,
    KEEP_BOTH,
}

File: ios-native/Jottery/Jottery/Models/ConflictInfo.swift
import Foundation

/// Holds both local and server versions of a conflicting note for resolution.
/// Decrypted content is stored for display; raw encrypted server data is stored for applying resolutions.
struct ConflictInfo: Identifiable, Sendable {
    let id: String  // noteId

    // Decrypted for display
    let localContent: String
    let localTags: [String]
    let localModifiedAt: String
    let localAttachments: [AttachmentRef]

    let serverContent: String
    let serverTags: [String]
    let serverModifiedAt: String

    // Encrypted server data for applying "Keep Server" / "Keep Both"
    let serverEncryptedContent: String
    let serverEncryptedTags: [String]
    let serverVersion: Int
    let serverAttachments: [AttachmentRef]
    let serverPinned: Bool
    let serverSyntaxLanguage: String?
    let serverWordWrap: Bool?
    let serverShowPreview: Bool?
    let serverContentHash: String?
    let serverParentHash: String?
    let serverHashChain: [String]?
}

enum ConflictResolutionStrategy: Sendable {
    case keepLocal
    case keepServer
    case keepBoth
}
